src/python/pants/base/workunit.py

```diff
@@ -146,9 +146,12 @@ def start(self, start_time=None):
   def end(self):
     """Mark the time at which this workunit ended."""
     self.end_time = time.time()
+
+    return self.path(), self.duration(), self._self_time(), self.has_label(WorkUnitLabel.TOOL)
+
+  def close_outputs(self):
```

Comment:
This function is pretty self explanatory but consider a brief docstring to comply with the rest of the file, e.g.
```python
"""Cleanup by closing all output streams."""
```

I'm not extremely familiar with what `outputs` entails, so that's where a brief high level overview would help me out when reading this code.